src/System.Private.Xml/tests/Misc/RandomizedHashing.cs

```diff
@@ -0,0 +1,88 @@
+        // - ACE and BDF pairs were searched brute force which really quickly gave results
+        private static readonly Tuple<string, string>[] _collidingStringsPairs = new Tuple<string, string>[]
+        {
+            new Tuple<string, string>("01`@a@", "10@`@a"),
+            new Tuple<string, string>("01Ps6t", "10sPt6"),
+            new Tuple<string, string>("AFdBbC", "FABdCb"),
+            new Tuple<string, string>("01xXUt", "10XxtU"),
+            new Tuple<string, string>("01Ps6t", "10sPt6"),
+            new Tuple<string, string>("12;E7J", "21E;J7"),
+            new Tuple<string, string>("024u3P", "20u4P3"),
+            new Tuple<string, string>("23`B8[", "32B`[8"),
+            new Tuple<string, string>("039X3q", "30X9q3"),
+            new Tuple<string, string>("23MoOl", "32oMlO"),
+            new Tuple<string, string>("45QOd;", "54OQ;d"),
+            new Tuple<string, string>("45HjrQ", "54jHQr"),
+            new Tuple<string, string>("46v:r<", "64:v<r")
+        };
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void StringsDontHashToAnyKnownNonRandomizedSets()
+        {
+            var setOfHashes = new Tuple<int, int, int>(_strings[0].GetHashCode(), _strings[1].GetHashCode(), _strings[2].GetHashCode());
+            Assert.False(_knownNonRandomizedHashesOfStrings.Contains(setOfHashes));
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void StringsDoNotUseAlgorithmSimilarToCoreClrWhenRandomizedHashingIsDisabled()
+        {
+            // Even though GetHashCode gives different results on .NET 4.6 and CoreCLR with disabled
+            // hash randomization those pairs are causing collisions on both platforms as they meet
+            // certain properties causing hash collisions
+            foreach (var collPair in _collidingStringsPairs)
+            {
+                Assert.NotEqual(collPair.Item1.GetHashCode(), collPair.Item2.GetHashCode());
+            }
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void XmlQualifiedNameUsesStringGetHashCode()
+        {
+            Assert.Equal("foo".GetHashCode(), new XmlQualifiedName("foo").GetHashCode());
+            Assert.Equal("asdf".GetHashCode(), new XmlQualifiedName("asdf").GetHashCode());
+            Assert.Equal("randhashing".GetHashCode(), new XmlQualifiedName("randhashing").GetHashCode());
+        }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)]
+        public void SecureStringHasherUsesStringGetHashCode()
+        {
+            Assert.Equal("foo".GetHashCode(), new SecureStringHasher().GetHashCode("foo"));
+            Assert.Equal("asdf".GetHashCode(), new SecureStringHasher().GetHashCode("asdf"));
+            Assert.Equal("randhashing".GetHashCode(), new SecureStringHasher().GetHashCode("randhashing"));
```

Comment:
This feels like an internal implementation detail to me... why do we care that it matches String.GetHashCode in value?